src/pages/workspaces/workspace/analysis/modal-utils.ts

```diff
@@ -0,0 +1,74 @@
+import _ from 'lodash/fp'
+import { cloudServices } from 'src/data/gce-machines'
+import { getToolLabelFromRuntime, PersistentDisk } from 'src/pages/workspaces/workspace/analysis/utils/tool-utils'
+
+import { defaultDataprocMachineType, getDefaultMachineType } from './runtime-utils'
+
+
+export interface IComputeConfig {
+    persistentDiskSize: number
+    persistentDiskType: {
+      label: string
+      displayName: string
+      regionToPricesName: string
+    }
+    masterMachineType: string
+    masterDiskSize: number
+    numberOfWorkers: number
+    numberOfPreemptibleWorkers: number
+    workerMachineType: string
+    workerDiskSize: number
+    componentGatewayEnabled: boolean
+    gpuEnabled: boolean
+    hasGpu: boolean
+    gpuType: string
+    numGpus: number
+    autopauseThreshold: number
+    computeRegion: string
+    computeZone: string
+}
+
+export const buildExistingEnvironmentConfig = (computeConfig:IComputeConfig, currentRuntimeDetails:any, currentPersistentDiskDetails:PersistentDisk, isDataproc:boolean) => {
```

Comment:
computeConfig and currentRuntimeDetails can probably have  meaningful types related to types in `models` folder